pilot/pkg/networking/core/v1alpha3/listener_waypoint.go

```diff
@@ -283,18 +293,22 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []WorkloadAndServices, svcs
 				PrefixLen:     cidr.PrefixLen,
 			})
 		}
-		chains = append(chains, directChain)
+		chains = append(chains, tcpChain, httpChain)
 		ipMatcher.RangeMatchers = append(ipMatcher.RangeMatchers,
 			&matcher.IPMatcher_IPRangeMatcher{
-				Ranges:  ipRange,
-				OnMatch: match.ToChain(directChain.Name),
+				Ranges: ipRange,
+				OnMatch: match.ToMatcher(match.NewAppProtocol(match.ProtocolMatch{
+					TCP:  match.ToChain(tcpChain.Name),
+					HTTP: match.ToChain(tcpChain.Name),
```

Comment:
```suggestion
					HTTP: match.ToChain(httpChain.Name),
```
I know its "WIP", but wasn't sure if it was intended

Reply:
It's breaking stuff :( Trying to isolate the failure.